runtime/src/dispatcher.rs

```diff
@@ -142,7 +153,20 @@ impl Dispatcher {
         Ok(())
     }
 
-    fn run(&self, initializer: Box<dyn Initializer>, rx: channel::Receiver<QueueItem>) {
+    /// Waits for dispatcher to abort.
+    pub fn wait_abort(&self) -> Result<()> {
```

Comment:
Maybe call it `abort_and_wait` (with doc comment like "Signals to dispatcher that it should abort and waits for the abort to complete.").